lib/main.dart

```diff
@@ -427,14 +426,44 @@ void addToAmplitude() async {
   var lambdaEndpt =
       'https://jsr2rwrw5m.execute-api.us-east-1.amazonaws.com/default/pb-lambda-microservice';
 
-  var body = json.encode({
+  var eventProperties = MainInfo().amplitudMap['eventProperties'];
+
+  eventProperties['eggs'] = PBPluginListHelper.names ?? [];
+
+  eventProperties['Design specification'] =
+      MainInfo().configuration.widgetStyle;
+
+  eventProperties['Percentage of Auto Layout as UI'] =
+      getPercentage('Number of auto layouts', ['Number of stacks']);
+
+  eventProperties['Percentage of design specification'] =
+      getPercentage('Number of theme text styles', ['Number of theme colors']);
+
+  MainInfo().amplitudMap.addAll({
     'id': MainInfo().deviceId,
-    'eventProperties': {'eggs': PBPluginListHelper.names ?? {}}
+    // 'id': 'Testing Analytics'
   });
 
+  var body = json.encode(MainInfo().amplitudMap);
+
   await http.post(
     Uri.parse(lambdaEndpt),
     headers: {HttpHeaders.contentTypeHeader: 'application/json'},
     body: body,
   );
 }
+
+String getPercentage(String property, List<String> properties) {
```

Comment:
This could also be a private method if we encapsulate all analytics-related stuff into a service to avoid scattering responsibilities